DEFEDIT-1171 Adapt build process for new editor-alpha channel

ragnard-king opened the pull request:
### Technical changes

- Only release_editor2 when channel is 'editor-alpha'. This channel
  does not exist yet, so until it does this means we won't be making
  any releases of editor2 on branches where this change has been
  merged.

- Introduce the notion of engine artifact version, can be either:
    - `dynamo-home` take everything from DYNAMO_HOME
    - `archived` download archived artifacts for HEAD
    - `archived-stable` download archived artifacts for VERSION-file
    - a sha - download archived artifacts for given sha

- bundle.py: Replace bundle arguments `--git-rev` and `--pack-local`
  with a single `--engine-artifacts` argument as above.

- Extend leiningen init task to support engine artifact version as an
  argument.
  - Additionally support an `auto` argument which will initialize
    editor project with artifacts depending on whether HEAD is
    descended from origin/dev (or origin/edutor-dev).

ragnard-king committed:
DEFEDIT-1171 Adapt build process for new editor-alpha channel

* Technical changes

- Only release_editor2 when channel is 'editor-alpha'. This channel
  does not exist yet, so until it does this means we won't be making
  any releases of editor2 on branches where this change has been
  merged.

- Introduce the notion of engine artifact version, can be either:
    - `dynamo-home` take everything from DYNAMO_HOME
    - `archived` download archived artifacts for HEAD
    - `archived-stable` download archived artifacts for VERSION-file
    - a sha - download archived artifacts for given sha

- bundle.py: Replace bundle arguments `--git-rev` and `--pack-local`
  with a single `--engine-artifacts` argument as above.

- Extend leiningen init task to support engine artifact version as an
  argument.
  - Additionally support an `auto` argument which will initialize
    editor project with artifacts depending on whether HEAD is
    descended from origin/dev (or origin/edutor-dev).

matgis-king reviewed (dismissed):
Some questions and concerns. Also, I guess we must lock down the Defold SDK version we use in the editor? This is currently tied to the editor version string we obtain from the `defold.sha1` property.

ragnard-king committed:
DEFEDIT-1171 Review fixes

- bundle.py:
  - fail when unable to find git sha from VERSION file
  - whitespace
- init task:
  - correct error message
  - add task docstring
  - use local dev/editor-dev when detecting what current ref is
    descended from.
- pack task:
  - simpler, correct docstring

ragnard-king committed:
DEFEDIT-1171 Review fixes

- separate editor/engine sha1 as they can now be different
- remove `do-init` alias with explicit list in `init` task
- fix incorrect error message in bundle.py

matgis-king reviewed (commented):
I still think `autodetect-sha` won't work in practice, even with the changes. We should also make sure the about box looks alright given that the row indices appear a bit weird. Other than that it looks good to me!

ragnard-king committed:
DEFEDIT-1171 Review fixes

- init task:
  - remove broken auto detection of sha
- slight improvement to about dialog

ragnard-king committed:
DEFEDIT-1171 Re-introduce engine artifact auto detection

- determine which engine sha to use depending on whether the contents
  of the VERSION file point to an existing branch or not.
- add ability to skip tests when bundling
